src/backend/distributed/planner/recursive_planning.c

```diff
@@ -1305,6 +1312,217 @@ ContainsReferencesToOuterQueryWalker(Node *node, VarLevelsUpWalkerContext *conte
 }
 
 
+/*
+ * WrapFunctionsInSubqueries iterates over all the immediate Range Table Entries
+ * of a query and wraps the functions inside (SELECT * FROM fnc() f) subqueries.
+ *
+ * We currently wrap only those functions that return a single value. If a
+ * function returns records or a table we leave it as it is
+ * */
+static void
+WrapFunctionsInSubqueries(Query *query)
+{
+	List *rangeTableList = query->rtable;
+	ListCell *rangeTableCell = NULL;
+
+	/* there needs to be at least two RTEs for a join operation */
+	if (list_length(rangeTableList) < 2)
```

Comment:
The comment could elaborate a bit about why we only care about joins.

Reply:
I added some explanations to why we only wrap functions that are joined. I did not go into so much detail, but I think this is enough